ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp

```diff
@@ -7125,6 +7125,13 @@ int OGRGeoPackageTableLayer::GetNextArrowArray(struct ArrowArrayStream* stream,
             {
                 break;
             }
+
+            // Install query logging callback
+            if ( task->m_poDS->pfnQueryLoggerFunc )
+            {
+                poOtherLayer->m_poDS->SetQueryLoggerFunc( task->m_poDS->pfnQueryLoggerFunc, task->m_poDS->poQueryLoggerArg );
+            }
```

Comment:
That's the reverse: you want to attach the callback of the main dataset (m_poDS) to the task dataset
```suggestion
            if ( m_poDS->pfnQueryLoggerFunc )
            {
                task->m_poDS->SetQueryLoggerFunc( m_poDS->pfnQueryLoggerFunc, m_poDS->poQueryLoggerArg );
            }
```